wfe/wfe.go

```diff
@@ -1775,6 +1846,91 @@ func (wfe *WebFrontEndImpl) orderForDisplay(
 	return result
 }
 
+// RenewalInfo implements ACME Renewal Info (ARI)
+func (wfe *WebFrontEndImpl) RenewalInfo(_ context.Context, response http.ResponseWriter, request *http.Request) {
+	if request.Method == http.MethodPost {
+		wfe.sendError(acme.InternalErrorProblem("POSTing to RenewalInfo has not been implemented yet"), response)
+		return
+	}
+
+	if len(request.URL.Path) == 0 {
+		wfe.sendError(acme.NotFoundProblem("Must specify a request path"), response)
+		return
+	}
+
+	certID, err := wfe.parseCertID(request.URL.Path)
+	if err != nil {
+		wfe.sendError(acme.MalformedProblem(fmt.Sprintf("Parsing ARI CertID failed: %s", err)), response)
+		return
+	}
+
+	renewalInfo, err := wfe.determineARIWindow(context.TODO(), certID)
+	if err != nil {
+		wfe.sendError(acme.InternalErrorProblem(fmt.Sprintf("Error determining renewal window: %s", err)), response)
+		return
+	}
+
+	response.Header().Set("Retry-After", strconv.Itoa(int(6*time.Hour/time.Second)))
+	err = wfe.writeJSONResponse(response, http.StatusOK, renewalInfo)
+	if err != nil {
+		wfe.sendError(acme.InternalErrorProblem(fmt.Sprintf("Error marshaling renewalInfo: %s", err)), response)
+		return
+	}
+}
+
+func (wfe *WebFrontEndImpl) determineARIWindow(_ context.Context, id *core.CertID) (*core.RenewalInfo, error) {
```

Comment:
Why take a context here at all?

```suggestion
func (wfe *WebFrontEndImpl) determineARIWindow(_ context.Context, id *core.CertID) (*core.RenewalInfo, error) {
```

Reply:
I've removed this one since it makes no sense in this, excuse my language, context.